pkg/errors/conversion.go

```diff
@@ -103,4 +104,7 @@ func setErrorDetails(err *Error, details ErrorDetails) {
 	if cause := details.Cause(); cause != nil {
 		err.cause, _ = From(cause)
 	}
+	if err.code == 0 {
```

Comment:
This doesn't seem consistent; checking if the target doesn't have a code set, while for other fields we check whether the source has the fields set

Reply:
```suggestion
	if code := details.Code(); code != 0 {
```

Reply:
That's what I thought as well initially, but that causes the `code` to be overwritten by whatever is in the details, which in turn causes panics and test failures all around.
I think we should just revert https://github.com/TheThingsNetwork/lorawan-stack/pull/976/commits/ffe7547b10695c83fe6f793d814a96c6089abe0f that @htdvisser 	added without my knowledge

Reply:
I simply moved it from the other file

Reply:
So what do we do with this?

Reply:
Removed @htdvisser 's commit, please review